cumulusci/tasks/debug_logs.py

```diff
@@ -0,0 +1,135 @@
+import datetime
+import os
+import csv
+
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+
+
+class TurnOnDebugLogs(BaseSalesforceApiTask):
+    task_docs = """
+    Turn on debugging information in Salesforce
+    """
+
+    debug_level_id = None  # Class variable
+
+    def _run_task(self):
+        """Create a TraceFlag for a given user."""
+        self._delete_debug_levels()
+        self._delete_trace_flags()
+        self.logger.info("Creating DebugLevel object")
+        DebugLevel = self._get_tooling_object("DebugLevel")
+        result = DebugLevel.create(
+            {
+                "ApexCode": "Info",
+                "ApexProfiling": "Debug",
+                "Callout": "Info",
+                "Database": "Info",
+                "DeveloperName": "CumulusCI",
+                "MasterLabel": "CumulusCI",
+                "System": "Info",
+                "Validation": "Info",
+                "Visualforce": "Info",
+                "Workflow": "Info",
+            }
+        )
+        if TurnOnDebugLogs.debug_level_id:
+            self.logger.info(
+                "Debug logs are already turned on ({TurnOnDebugLogs.debug_level_id})"
+            )
+            return
+        TurnOnDebugLogs.debug_level_id = result["id"]
+        self.logger.info("Setting up trace flag to capture debug logs")
+        # New TraceFlag expires 12 hours from now
+        expiration_date = datetime.datetime.utcnow() + datetime.timedelta(
+            seconds=60 * 60 * 12
+        )
+        TraceFlag = self._get_tooling_object("TraceFlag")
+        result = TraceFlag.create(
+            {
+                "DebugLevelId": result["id"],
+                "ExpirationDate": expiration_date.isoformat(),
+                "LogType": "USER_DEBUG",
+                "TracedEntityId": self.org_config.user_id,
+            }
+        )
+        self.logger.info("Created TraceFlag for user")
+
+    def _delete_trace_flags(self):
+        """
+        Delete existing DebugLevel objects.
+        This will automatically delete associated TraceFlags as well.
+        """
+        self.logger.info("Deleting existing TraceFlag objects")
+        result = self.tooling.query(
+            "Select Id from TraceFlag Where TracedEntityId = '{}'".format(
+                self.org_config.user_id
+            )
+        )
+        if result["totalSize"]:
+            TraceFlag = self._get_tooling_object("TraceFlag")
+            for record in result["records"]:
+                TraceFlag.delete(str(record["Id"]))
+
+    def _delete_debug_levels(self):
+        """
+        Delete existing DebugLevel objects.
+        This will automatically delete associated TraceFlags as well.
+        """
+        self.logger.info("Deleting existing DebugLevel objects")
+        result = self.tooling.query("Select Id from DebugLevel")
```

Comment:
We should delete only our own DebugLevel by DeveloperName. Users may have created their own Debug Levels that they don't want smashed if they run this task locally.

Reply:
Do we run a similar risk inside of `_delete_trace_flag()` where the query is selecting all flags where the `TracedentityId == self.org_config.user_id`? Is there the potential that some of those flags were not created by this task?

Reply:
Yeah I think we need to be more careful to not destroy debug levels / trace flags that came from something else.